api/agent.go

```diff
@@ -353,7 +353,13 @@ func NewControllerWrapper(config *loadtest.Config, controllerConfig interface{},
 		if metrics != nil {
 			ueSetup.Metrics = metrics.UserEntityMetrics()
 		}
-		ue := userentity.New(ueSetup, ueConfig)
+		var ue *userentity.UserEntity
+		if index == 0 {
+			// We want a user to be admin in order to be able to perform admin allowed actions
+			ue = admin(ueSetup, ueConfig, config.ConnectionConfiguration.AdminEmail, config.ConnectionConfiguration.AdminPassword)
+		} else {
+			ue = userentity.New(ueSetup, ueConfig)
+		}
```

Comment:
Why the additional parameter? I thought adding a check on `id == 0` would be sufficient.

This problem should be really dealt with during data initialization by making some (configurable) percentage of the users to be admins. We can do this at a later time of course. I am okay with the quick fix here.

Reply:
I thought the userId is not guaranteed to start from 0, since we randomize but I see that we do that only after the first user has been initialized: https://github.com/mattermost/mattermost-load-test-ng/blob/master/loadtest/loadtest.go#L99.

> This problem should be really dealt with during data initialization by making some (configurable) percentage of the users to be admins. We can do this at a later time of course. I am okay with the quick fix here.

Completely agree with this, i am ok with proceeding with that approach here.

Reply:
> I thought the userId is not guaranteed to start from 0, since we randomize but I see that we do that only after the first user has been initialized: https://github.com/mattermost/mattermost-load-test-ng/blob/master/loadtest/loadtest.go#L99.

Yes, also that's only done if we allow multiple sessions per user, which is not usually the case.

> Completely agree with this, i am ok with proceeding with that approach here.

That would be even better :+1:

Reply:
Sorry i focused yesterday on the message export work, have a look on https://github.com/mattermost/mattermost-load-test-ng/pull/412/commits/f78df399e6b31e0c4d3e30f03f68a91878ef8c41 , the `loadtest` module itself kknows about the index of the user that is adding, hence having all the `addUser` and `addUser(s)` methods init. So I figured a deterministic way to identify if we need an admin is to pass bool argument from loadtest to agent. 

The addUsers now is asymmetric with the removeUsers happy to make removeUsers to remove the right % of admin vs non-admin users